### ISSUE TITLE
Postgres recovery issue, possibly caused by lack of garbage collection on jobs listing 

### ISSUE THREAD

**Issue by temujin9** [flynn/flynn#3413]
A test cluster produced a 12 hour long Postgres degradation alert, via monitoring of the status app. This appeared to coincide with an earlier ramp up to 100% CPU usage (but low load) on one node, which was traced to the flynn-host daemon and the flynn-scheduler container. That scheduler job was listed as down for a long time (via `flynn ps -a`) with no attempt to replace it visible as it normally would be on that list. 

Running `flynn-host collect-debug-info` on that node resulted in a huge load spike on the wedged node, an OOM partway through the collection, a subsequent drop in CPU usage, and the a clearing of the pg status alert. This all coincided relatively closely, making me think the symptoms are related despite broadly different start times.

Also probably salient is that there were roughly 80K of terminated jobs (via `flynn-host ps -a`), from an old flapping job which had (a while before this) been turned off.

https://botbot.me/freenode/flynn/2016-09-21/?msg=73387396&page=1 has more of the conversation around debugging it, including recovered logs and the process list.

**Comment by temujin9** [comment]
Additional note: the lack of status alert on the wedging of the scheduler is potentially more problematic, in my book. We should have known it was having issues back when they started, not after they cascaded.

**Comment by temujin9** [comment]
Hrm, okay it appears I've misinterpreted the scheduler stuff. It looks like there's supposed to be only one, not two like I'm seeing via `flynn -a controller ps --all`:

```
ID                                                 TYPE       STATE  CREATED      RELEASE
ip1010036184-e05fa18c-606d-4af1-97be-d8940c3bc8c6  scheduler  up     2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036184-1ba1eaf8-f2df-4b8c-9f14-ca577ddf9809  web        up     2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036183-332d8377-a347-4150-ae2d-13588664a38f  scheduler  up     2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036185-102abd33-acad-43c4-9ac7-cbfe87bcd986  scheduler  down   2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036183-4739c891-10a8-471d-a452-a784d94413c0  worker     up     2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036183-1ce93699-efdb-4c52-9c01-cf0d0a6dde6f  web        up     2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036184-53b7774f-e1bd-4f61-b400-33e0fd360312  worker     up     2 weeks ago  0fca8fb7-2ee0-43be-8c63-586e8fe972c2
ip1010036183-871bdcca-442f-4fbc-be4b-237c8dd64243  run        down   6 days ago   4f004ffb-832b-4a8d-a8de-3f53b70205ad
ip1010036184-ea0ebbb6-5240-4db3-9800-f24f3ff753e2  run        down   6 days ago   4f004ffb-832b-4a8d-a8de-3f53b70205ad
ip1010036185-81ace49d-456c-45b5-89c1-105451fad3d9  run        down   6 days ago   4f004ffb-832b-4a8d-a8de-3f53b70205ad
```

**Comment by temujin9** [comment]
Ah, no: I was right, that's an omni job. So: the third scheduler is still down, not up like I thought. :-/

**Comment by lmars** [comment]
Our current thinking here is for flynn-host to only track the last N stopped jobs, and I am going to run some benchmarks to determine an appropriate value for N.

I think the subsequent issues (a scheduler being down, postgres becoming unhealthy etc.) were all due to the high load caused by the schedulers and flynn-hosts stuck in a job sync feedback loop.

**Comment by lmars** [comment]
Fixed in #3767.
